### PR TITLE
Converts svg to png (bgcolor, opacity, text)

### PR DESCRIPTION
**Pull request by pushkalkatara**
Fixes #
https://github.com/fossasia/badgeyay/issues/155
[Enhancement] Option to user to select background, bg color, opacity, text and generate badge.
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Generating PNGs according to user selections and printing badges.
- User can select Background, Background Color, Opacity, Text.

##### Suggestion:

-Shifting to process to generate pngs through svgs for modularity rather then selecting pre-built pngs.
-This could solve problems like size of badges etc, all the problems related to design of badges.

### CODE REVIEW

**Comment by pushkalkatara** [comment]
Also the svg in the previous merge seems buggy. It is not generating fossasia logo on the top and the name. @gabru-md

File: app/main.py [hunk 1]
```diff
@@ -5,15 +5,18 @@
...
 APP_ROOT = os.path.dirname(os.path.abspath(__file__))
 UPLOAD_FOLDER = os.path.join(APP_ROOT, 'static/uploads')
+SVGS_FOLDER = os.path.join(APP_ROOT, 'static/svgs')
 BADGES_FOLDER = os.path.join(APP_ROOT, 'static/badges')
 SCRIPT = os.path.join(APP_ROOT, 'merge_badges.sh')
 
 app = Flask(__name__)
 app.config['DEBUG'] = True
 app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+app.config['SVGS_FOLDER'] = SVGS_FOLDER
```
Comment (jajodiaraghav) [inline app/main.py]: Please remove this line, I don't think we need this!

File: app/main.py [hunk 2]
```diff
@@ -74,7 +78,13 @@ def upload():
...
 	# If default background is selected
 	if img != '':
-		filename = img + '.csv'
+		if(img == 'user_defined.png'):
+			bg_color = request.form['bg_color']
+			text_on_image = request.form['text_on_image']
+			do_svg2png(img, 1, bg_color, text_on_image)
+			filename = img + '.csv'
+		else:
+			filename = img + '.csv'
```
Comment (jajodiaraghav) [inline app/main.py]: Remove the else part and put `filename = img + '.csv'` outside the if block.

**Comment by rupav** [comment]
Hi @pushkalkatara, found the problem. I was seeing the images in INKSCAPE only. When i opened it in browser, it was not showing completely:
not_browser_supported
Now its showing correctly, by correcting the document properties in the inkscape.
.

@pushkalkatara please send your email. I will send the rectified source files. You can then commit all in one. 
Sorry once again for the incovenience :(.

File: app/templates/index.html [hunk 3]
```diff
@@ -88,6 +88,9 @@
 							</li>
 							{% endfor %}
 						</ul>
+						<input id="cp1" type="text" class="form-control" value="Enter Color HEX if user_defined" name="bg_color"/>
+						<!---><script> $(function(){$('#cp1').colorpicker();});</script><Idea is to Add Color Picker which generates HEX vales--->
```
Comment (jajodiaraghav) [inline app/templates/index.html]: Please remove this comment line

File: app/templates/index.html [hunk 4]
```diff
@@ -88,6 +88,9 @@
 							</li>
 							{% endfor %}
 						</ul>
+						<input id="cp1" type="text" class="form-control" value="Enter Color HEX if user_defined" name="bg_color"/>
```
Comment (jajodiaraghav) [inline app/templates/index.html]: `Enter Color HEX if user_defined` I think this should be in a placeholder

File: app/templates/index.html [hunk 5]
```diff
@@ -88,6 +88,9 @@
...
+						<!---><script> $(function(){$('#cp1').colorpicker();});</script><Idea is to Add Color Picker which generates HEX vales--->
+						<input id="image_text" type="text" class="form-control" value="Enter Text if user_defined" name="text_on_image"/>
```
Comment (jajodiaraghav) [inline app/templates/index.html]: Placeholder

**Comment by jajodiaraghav** [comment]
@pushkalkatara You also pushed the residue files in badges folder, remove them.
Also Please remove the sample svg and image file that you uploaded

File: app/main.py (hunk 1 above)
Comment (pushkalkatara) [reply app/main.py]: This may be needed afterwards. but for now i'm removing this

**Comment by pushkalkatara** [comment]
@jajodiaraghav Yup, i uploaded the residue files to show the error in svg, i removed them in the update.

**Comment by jajodiaraghav** [comment]
@pushkalkatara You do not put residue files in a pull request! If you want to show files, you upload it directly on issue thread. Please read about best practices and "How to create a CLEAN PR".

**Comment by pushkalkatara** [comment]
Okay. I've updated the svg from rupav and now its printing fine.

**Comment by gabru-md** [comment]
> Okay. I've updated the svg from rupav and now its printing fine.

@pushkalkatara can you share a screenshot for this?
Thanks!

**Comment by pushkalkatara** [comment]
@gabru-md Its in the commit under the file change section, you can check it.

**Comment by gabru-md** [comment]
@jajodiaraghav LGTM.
Please review and let me know if it can be merged.:+1:

**Comment by jajodiaraghav** [comment]
@gabru-md Please merge 🚀 
This looks good to me.

Thanks a lot @pushkalkatara  :)

**Comment by pushkalkatara** [comment]
Thanks for the merge :D

**Pull request merged**